lib/jsonapi_parameters/stack_limit.rb

```diff
@@ -0,0 +1,45 @@
+module JsonApi
+  module Parameters
+    LIMIT = 3
+
+    class StackLevelTooDeepError < StandardError
+    end
+
+    def stack_limit=(val)
+      @stack_limit = val
+    end
+
+    def stack_limit
+      @stack_limit || LIMIT
+    end
+
+    def reset_stack_limit
+      @stack_limit = LIMIT
+    end
+
+    private
+
+    def initialize_stack(custom_stack_limit)
+      @current_stack_level = 0
+      @stack_limit = custom_stack_limit
```

Comment:
how about we put 
```
@stack_limit = custom_stack_limit || LIMIT
```
in here. Then the getter and the setter could be totally pure and have no additional logic.

Reply:
They won't have no additional logic, as in different places we also use these getters to pre-initialize the value in case its empty. 
https://github.com/visualitypl/jsonapi_parameters/pull/43/files#diff-1589ffd377f1f2d9e84dc1e63953f874R6

Reply:
Since we `#initialize_stack` inside `#jsonapify`, then `stack_limit` getter will be `nil` in the default scenario. However in the future we still want to use this getter to ensure using custom limit if it was provided before.

Reply:
ok, thanks!